feat: migrate SlowBuffer to Buffer.allocUnsafeSlow() (#193)
nodejs/userland-migrations: recipes/slow-buffer-to-buffer-alloc-unsafe-slow/README.md

# SlowBuffer to Buffer.allocUnsafeSlow Codemod

> This codemod migrates deprecated `SlowBuffer` usage to `Buffer.allocUnsafeSlow()` to handle Node.js DEP0030.

Reviewer: ```suggestion
This codemod migrates deprecated `SlowBuffer` usage to `Buffer.allocUnsafeSlow()` to handle Node.js [DEP0030](https://nodejs.org/api/deprecations.html#DEP0030).
```

We link to the doc so when user read this readme on [Codemod Registry](https://app.codemod.com/registry) he can easily access to the deprecation.

## What it does

This codemod transforms:

> 1. `SlowBuffer` constructor calls to `Buffer.allocUnsafeSlow()`
> 2. Direct `SlowBuffer` calls to `Buffer.allocUnsafeSlow()`
> 3. Import/require statements to use `Buffer` instead of `SlowBuffer`

Reviewer: ```suggestion
-`SlowBuffer` constructor calls to `Buffer.allocUnsafeSlow()`
- Direct `SlowBuffer` calls to `Buffer.allocUnsafeSlow()`
- Import/require statements to be synced with new function
```

it's different point there are any kind order. 

Therefore, it does not necessarily transform `Buffer` into `SlowBuffer`; it depends on the case.

> ## Examples

Reviewer: example on rest of codemods use this structure :

```md
## Examples

**Before:**
<!-- code snippet  -->

**After:**

<!-- code snippet  -->
```

I think your aproach is better in terms of semantics.

@nodejs/userland-migrations what should we do modify this pr or update other codemod ?

### Before

```javascript
const SlowBuffer = require("buffer").SlowBuffer;

// Using SlowBuffer constructor
const buf1 = new SlowBuffer(1024);
const buf2 = SlowBuffer(512);

// Direct import
const { SlowBuffer } = require("buffer");
const buf3 = new SlowBuffer(256);
```

### After

```javascript
const Buffer = require("buffer").Buffer;

// Using Buffer.allocUnsafeSlow()
const buf1 = Buffer.allocUnsafeSlow(1024);
const buf2 = Buffer.allocUnsafeSlow(512);

// Direct import
const { Buffer } = require("buffer");
const buf3 = Buffer.allocUnsafeSlow(256);
```

> ### ESM Support

Reviewer: we usually have one example in ESM and CJS is simplicit.

```javascript
// Before
import { SlowBuffer } from "buffer";
const buf = new SlowBuffer(1024);

// After
import { Buffer } from "buffer";
const buf = Buffer.allocUnsafeSlow(1024);
```

> ## References
> 
> - [Node.js DEP0030: SlowBuffer](https://nodejs.org/api/deprecations.html#DEP0030)
> - [nodejs/node#58220](https://github.com/nodejs/node/pull/58220)

Reviewer: ```suggestion
```

No needed because you are making the link on the introduction and the pr isn't really useful in codemod context.
